src/routes/Forecast.svelte

```diff
@@ -1,7 +1,41 @@
 <script>
+  import ForecastMainBanner from '../components/ForecastMainBanner.svelte';
+  import weatherService from '../services/weatherService';
+
   export let params;
+
+  let weatherData = {};
+
+  $: if (params.latitude && params.longitude) {
+    (async () => {
```

Comment:
Not sure about using IIFE here, it potentially can be a reason of memory leak. Since the IIFE is defined within the reactive statement, it captures the context of `weatherData` and other variables. If it were to retain a reference to these variables beyond its execution, it could lead to memory leaks. But not sure @oleksandrstarov what do you think about it?